src/workspace/discovery.rs

```diff
@@ -31,348 +33,443 @@ pub enum DenoOrPkgJson {
+              member_url: member_dir_url,
             });
           }
-          // try to find the package.json in memory from what we already
+          Ok(member_dir_url)
+        };
+      let mut find_member_config_folder =
+        |raw_member: &str,
+         member_dir_url: &Url|
+         -> Result<_, ResolveWorkspaceMemberError> {
+          // try to find the config folder in memory from the configs we already
           // found on the file system
-          if let Some(pkg_json) = found_configs.remove(&pkg_json_path) {
-            return Ok(pkg_json);
+          if let Some(config_folder) =
+            found_config_folders.remove(member_dir_url)
+          {
+            return Ok(config_folder);
           }
 
-          // now search the file system
-          PackageJson::load_from_path(
-            &pkg_json_path,
-            opts.fs,
-            opts.pkg_json_cache,
-          )
-          .map_err(|e| e.into())
+          let maybe_config_folder = load_config_folder(&normalize_path(
+            root_config_file_path.join(raw_member),
+          ))?;
+          maybe_config_folder.ok_or_else(|| {
+            ResolveWorkspaceMemberError::NotFound {
+              dir_url: member_dir_url.clone(),
+            }
+          })
         };
+      let mut add_member = |raw_member: &str,
+                            member_dir_url: Url,
+                            member_config_folder: ConfigFolder|
+       -> Result<(), ResolveWorkspaceMemberError> {
+        let previous_member =
+          final_members.insert(new_rc(member_dir_url), member_config_folder);
+        if previous_member.is_some() {
+          Err(ResolveWorkspaceMemberError::Duplicate {
+            member: raw_member.to_string(),
+          })
+        } else {
+          Ok(())
+        }
+      };
+      if let Some(deno_json) = root_config_folder.deno_json() {
+        if let Some(members) = &deno_json.json.workspace {
+          if members.is_empty() {
+            return Err(
+              WorkspaceDiscoverErrorKind::MembersEmpty(
+                deno_json.specifier.clone(),
+              )
+              .into(),
+            );
+          }
+          for raw_member in members {
+            let member_dir_url = resolve_member_url(raw_member)?;
+            let member_config_folder =
+              find_member_config_folder(raw_member, &member_dir_url)?;
+            add_member(raw_member, member_dir_url, member_config_folder)?;
+          }
+        }
+      }
+      if let Some(pkg_json) = root_config_folder.pkg_json() {
+        if let Some(members) = &pkg_json.workspaces {
+          let mut has_warned = false;
+          for raw_member in members {
+            if raw_member.contains('*') {
+              if !has_warned {
+                has_warned = true;
+                log::warn!(
+                  "Wildcards in npm workspaces are not yet supported. Ignoring."
+                );
```

Comment:
Not a big deal, but without emoji/color this might be easy to miss.

Reply:
Let's just leave this as-is for now because hopefully we can implement this before the next minor release. I opened https://github.com/denoland/deno_config/issues/67